firefox-ios/Providers/Pocket/PocketProvider.swift

```diff
@@ -66,49 +65,56 @@ class PocketProvider: PocketStoriesProviding, FeatureFlaggable, URLCaching {
-        if shouldUseMockData {
-            return getMockDataFeed(count: items, completion: completion)
-        }
+            // Check if we should use mock data
+            if shouldUseMockData {
+                return try await getMockDataFeed(count: items)
+            }
 
-        // Ensure the feature is enabled and current locale is supported
-        guard isFeatureEnabled, isCurrentLocaleSupported else {
-            completion(.failure(Error.failure))
-            return
-        }
+            // Ensure the feature is enabled and current locale is supported
+            guard isFeatureEnabled, isCurrentLocaleSupported else {
+                throw Error.failure
+            }
 
-        // Note: Global feed is restricted to specific locale and feature availability
-        getGlobalFeed(items: items, completion: completion)
+            // Note: Global feed is restricted to specific locale and feature availability
+            return try await getGlobalFeed(items: items)
+        } catch {
+            throw error
+        }
     }
 
-    private func getGlobalFeed(items: Int, completion: @escaping (StoryResult) -> Void) {
+    private func getGlobalFeed(items: Int) async throws -> [PocketFeedStory] {
         guard let request = createGlobalFeedRequest(items: items) else {
-            return completion(.failure(Error.failure))
+            throw Error.failure
         }
 
         if let cachedResponse = findCachedResponse(for: request),
            let items = cachedResponse["recommendations"] as? [[String: Any]] {
-            return completion(.success(PocketFeedStory.parseJSON(list: items)))
+            return PocketFeedStory.parseJSON(list: items)
         }
 
-        urlSession.dataTask(with: request) { (data, response, error) in
-            guard let response = validatedHTTPResponse(response, contentType: "application/json"),
-                  let data = data
-            else { return completion(.failure(Error.failure)) }
+        do {
+            let (data, response) = try await urlSession.data(for: request)
+
+            guard let response = validatedHTTPResponse(response, contentType: "application/json") else {
+                throw Error.failure
+            }
 
             self.cache(response: response, for: request, with: data)
 
-            let json = try? JSONSerialization.jsonObject(with: data, options: .allowFragments) as? [String: Any]
+            let json = try JSONSerialization.jsonObject(with: data, options: .allowFragments) as? [String: Any]
             guard let items = json?["recommendations"] as? [[String: Any]] else {
-                return completion(.failure(Error.failure))
+                throw Error.failure
             }
 
-            return completion(.success(PocketFeedStory.parseJSON(list: items)))
-        }.resume()
+            return PocketFeedStory.parseJSON(list: items) // Added missing return
+        } catch {
+            throw error
+        }
```

Comment:
Same nit: This is a throwing method so you don't need the extra top level `do { ... } catch { throw error }`

Reply:
Sorry :( got busy and missed these, pushed now